joomla/joomla-cms
# Return all viewlevels for superusers

As discussed in https://github.com/joomla/joomla-cms/pull/16411#issuecomment-305423849

### Summary of Changes
This changes `JAccess::getAuthorisedViewLevels()` to return all viewlevels in case the user is Super User (has global core.admin permissions).
At the same time, it simplifies the check for root users as $user->authorise already checks for that.

### Testing Instructions
As long as https://github.com/joomla/joomla-cms/pull/16411 isn't merged, the same check could be done. This is basically an alternative way of fixing it.

### Documentation Changes Required
This is a different behavior of the method from today, however since the root user already has that behavior, it imho would make sense to threat super users the same as root users.
